torch/csrc/api/include/torch/nn/modules/any.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+
+#include <torch/detail/any.h>
+
+#include <torch/csrc/autograd/variable.h>
+#include <torch/csrc/utils/memory.h>
+#include <torch/csrc/utils/variadic.h>
+
+#include <memory>
+#include <type_traits>
+#include <typeinfo>
+#include <utility>
+#include <vector>
+
+namespace torch {
+namespace detail {
+template <typename Head = void, typename... Tail>
+void check_not_lvalue_references() {
+  static_assert(
+      !std::is_lvalue_reference<Head>::value ||
+          std::is_const<typename std::remove_reference<Head>::type>::value,
+      "Modules stored inside AnyModule must not take (lvalue) references. Use pointers instead.");
```

Comment:
Why not?

This error feels like it would be totally meaningless to me if I was not very well-versed the vagaries of C++ references.

Reply:
Because in this mechanism we are copying/movying parameters into `Any` objects that live somwhere on the stack, so it would be inherently unsafe to allow users to take lvalue refs to these values, especially since the values are different from the ones they pass into the function call! Should I simplify the error message and add a comment on top?